qmi/instruments/picotech/picoscope3403.py

```diff
@@ -177,46 +215,49 @@ def acquire_by_trigger_and_get_data(
 
             self.set_channel(chan, True, couplings[chan], ranges, 0)
 
-        number_samples = int(time_span / time_base)
+        # Calculate the total number of samples in the time span.
+        number_samples = time_span // sampling_interval
+        _logger.info(f"Timespan = %i Num samples = %i", time_span, number_samples)
```

Comment:
logger should use lazy formatting instead of f-string

Reply:
REmoved the 'f'